api/src/test/java/io/opencensus/common/ServerStatsEncodingTest.java

```diff
@@ -0,0 +1,115 @@
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit tests for {@link ServerStatsEncoding}. */
+@RunWith(JUnit4.class)
+public class ServerStatsEncodingTest {
+
+  @Test
+  public void encodeDecodeTest() {
+    ServerStats serverStatsToBeEncoded = null;
+    ServerStats serverStatsDecoded = null;
+    byte[] serialized = null;
+
+    serverStatsToBeEncoded = ServerStats.create(31, 22, (byte) 1);
+    serialized = ServerStatsEncoding.toBytes(serverStatsToBeEncoded);
+    serverStatsDecoded = ServerStatsEncoding.parseBytes(serialized);
+    assertThat(serverStatsDecoded).isEqualTo(serverStatsToBeEncoded);
+
+    serverStatsToBeEncoded = ServerStats.create(0, 22, (byte) 0);
+    serialized = ServerStatsEncoding.toBytes(serverStatsToBeEncoded);
+    serverStatsDecoded = ServerStatsEncoding.parseBytes(serialized);
+    assertThat(serverStatsDecoded).isEqualTo(serverStatsToBeEncoded);
+
+    serverStatsToBeEncoded = ServerStats.create(450, 0, (byte) 0);
+    serialized = ServerStatsEncoding.toBytes(serverStatsToBeEncoded);
+    serverStatsDecoded = ServerStatsEncoding.parseBytes(serialized);
+    assertThat(serverStatsDecoded).isEqualTo(serverStatsToBeEncoded);
+  }
+
+  @Test
+  public void skipUnknownFieldTest() {
+    ServerStats serverStatsToBeEncoded = null;
+    ServerStats serverStatsDecoded = null;
+    byte[] serialized = null;
+
+    serverStatsToBeEncoded = ServerStats.create(31, 22, (byte) 1);
+    serialized = ServerStatsEncoding.toBytes(serverStatsToBeEncoded);
+
+    // Add new field at the end.
+    byte[] serializedExpanded = new byte[serialized.length + 9];
+    System.arraycopy(serialized, 0, serializedExpanded, 0, serialized.length);
+    final ByteBuffer bb = ByteBuffer.wrap(serializedExpanded);
+    bb.order(ByteOrder.LITTLE_ENDIAN);
+    bb.position(serialized.length);
+    bb.put((byte) 255);
+    bb.putLong(0L);
+    byte[] newSerialized = bb.array();
+
+    serverStatsDecoded = ServerStatsEncoding.parseBytes(newSerialized);
+    assertThat(serverStatsDecoded).isEqualTo(serverStatsToBeEncoded);
+  }
+
+  @Test
+  public void negativeLbLatencyValueTest() {
+    ServerStats serverStatsToBeEncoded = null;
+    ServerStats serverStatsDecoded = null;
+    byte[] serialized = null;
+
+    serverStatsToBeEncoded = ServerStats.create(31, 22, (byte) 1);
+    serialized = ServerStatsEncoding.toBytes(serverStatsToBeEncoded);
+
+    // update serialized byte[] with negative value for lbLatency.
+    final ByteBuffer bb = ByteBuffer.wrap(serialized);
+    bb.order(ByteOrder.LITTLE_ENDIAN);
+    bb.position(2);
+    bb.putLong(-100L);
+
+    byte[] newSerialized = bb.array();
+    serverStatsDecoded = ServerStatsEncoding.parseBytes(newSerialized);
+    assertThat(serverStatsDecoded).isNull();
+  }
+
+  @Test
+  public void negativeServerLatencyValueTest() {
```

Comment:
It would be helpful to add another test on invalid version.

Reply:
done.